errorDashboardJS/src/apiClient.ts

```diff
@@ -0,0 +1,107 @@
+import { customFetch } from "./fetch";
+import { configs, type Configs } from "./configs";
+import { baseUrl } from "./environment";
+import type { CreateErrorRequestType, CreateTagRequestType } from "./types";
+
+interface InitializeClient {
+  clientId: string;
+  clientSecret: string;
+  window: Window;
+}
+
+export class ErrorDashboardClient {
+  private clientId: string;
+  private clientSecret: string;
+  private window: Window;
+  private configs: Configs;
+
+  constructor(obj: InitializeClient) {
+    this.clientId = obj.clientId;
+    this.clientSecret = obj.clientSecret;
+    this.window = obj.window;
+    this.configs = configs;
+  }
+
+  async sendError(
+    error: Error,
+    message: string,
+    statusCode?: number,
+    tags?: CreateTagRequestType[]
+  ) {
+    let errorStack: string | undefined;
+    let errorLineNum: number | undefined;
+    let errorPath: string | undefined;
+    let errorUserAgent: string;
+
+    try {
+      errorStack = error.stack || "Error stack not found";
+      const stackLines = errorStack.split("\n");
+      if (stackLines.length > 1) {
+        // Match on first at... EG: at {filename}:{line}:{column #}
+        const match = stackLines[1].match(/at\s+(.*):(\d+):(\d+)/);
+        if (match) {
+          errorPath = match[1];
+          errorLineNum = parseInt(match[2], 10);
+        } else {
+          errorPath = "Error path not found";
+          errorLineNum = 0;
```

Comment:
Does `errorPath` and/or `errorLineNum` require special treatment like this, and if so, should it happen client-side?

Reply:
Yeah this can probably be parsed in the backend